src/main/java/org/spongepowered/common/mixin/core/scoreboard/MixinScoreboardLogic.java

```diff
@@ -213,9 +213,8 @@ public void removeObjective(ScoreObjective objective) {
         return Optional.empty();
     }
 
-    @SuppressWarnings("unchecked")
     public Set<Team> scoreboard$getTeams() {
-        return new HashSet<>((Set) this.teams.values());
+        return this.teams.values().stream().map(team -> (Team) team).collect(Collectors.toSet());
```

Comment:
I've seen that they've argumented this cast is never valid on the Eclipse bug tracker, however your replaced method is redundant because `ScorePlayerTeam` implements `Team` **only at runtime** due to our use of Mixin. Therefore the previous cast was actually perfectly safe and much more efficient.